### PR TITLE
Implement Error for ExactlyOneError, change interals of it a bit

### LINKED ISSUE #482: Should 'ExactlyOneError' implement std-'Error'?
The `exactly_one()` function was added in https://github.com/rust-itertools/itertools/pull/310:
```rust
exactly_one(self) -> Result<Self::Item, ExactlyOneError<Self>> 
```

For using the early return operator `?` on its result, it would be nice if `ExactlyOneError` would implement `Error`. Or is there any reason not to?

cc @Xaeroxe @Centril @bluss

### PR DESCRIPTION
**Pull request by Xaeroxe**
Fixes #482

### CODE REVIEW

**Comment by jswrenn** [comment]
Looks like `Error` is std-only.

**Comment by Xaeroxe** [comment]
CI failure is unrelated to this PR

File: src/ziptuple.rs [hunk 1]
```diff
@@ -108,7 +108,7 @@ macro_rules! impl_zip_iter {
             #[inline]
             fn next_back(&mut self) -> Option<Self::Item> {
                 let ($(ref mut $B,)*) = self.t;
-                let size = *[$( $B.len(), )*].into_iter().min().unwrap();
```
Comment (jswrenn) [inline src/ziptuple.rs:111]: I think you're inadvertently reverting #479. Could you rebase?

**Review by jswrenn — CHANGES_REQUESTED** [review]

File: src/ziptuple.rs (hunk 1 above)
Comment (Xaeroxe) [reply src/ziptuple.rs:111]: Sure thing!

**Comment by Xaeroxe** [comment]
@jswrenn this is ready for review.

**Review by jswrenn — APPROVED** [review]
bors r+

**Pull request merged**